messaging/src/androidTest/java/io/lantern/messaging/MessagingTest.kt

```diff
@@ -2117,6 +2123,56 @@ class MessagingTest : BaseMessagingTest() {
         }
     }
 
+    @Test
+    fun testRecovery() {
+        testInCoroutine {
+            newDB.use { dogDB ->
+                newDB.use { cat1DB ->
+                    newDB.use { cat2DB ->
+                        newMessaging(dogDB, "dog").with { dog ->
+                            newMessaging(cat1DB, "cat").with { cat1 ->
```

Comment:
NIT

```suggestion
                            newMessaging(cat1DB, "cat1").with { cat1 ->
```

and change the other to cat2, just for the reader to be sure this name doesn't affect the outcome